mathics/builtin/string/regexp.py

```diff
@@ -24,9 +24,9 @@ class RegularExpression(Builtin):
      = RegularExpression[[abc]]
 
     ## Mathematica doesn't seem to verify the correctness of regex
-    #> StringSplit["ab23c", RegularExpression["[0-9]++"]]
-     : Element RegularExpression[[0-9]++] is not a valid string or pattern element in RegularExpression[[0-9]++].
-     = StringSplit[ab23c, RegularExpression[[0-9]++]]
+    #> StringSplit["ab23c", RegularExpression["[0-9]++)"]]
```

Comment:
Why these changes?

Reply:
Because the previous pattern seems to be understood as a valid pattern by the new 
version of the regex parser

Reply:
Ok. I guess we should fix that.